### DISCUSSION TITLE
Combine 2 separate filter fields into one sql query

### DISCUSSION THREAD

**Discussion by Fak3** [vitalik/django-ninja#1307] [Q&A]
My frontend filtering form sends separate date and time fields. On the backend I need to validate both fields, merge them into one sql query Q(timestamp__gt=\<datetime\>), and exclude the separate date and time fields in sql query.

I wonder how I can achieve that without resorting to writing the whole custom_expression() query?

**Comment by Fak3** [comment]
The closest way I found still includes building custom expression. And uses custom list subclass hack to exclude date and time fields from filtering:

```python

class Exclude(list):
    """ 
    A list which instance is always True, even if empty. 
    Tricks ninja.FilterSchema to skip pydantic field from generating queryset.
    """
    def __bool__(self):
        return True

class Filters(FilterSchema):
    date__gt: Optional[date | Literal["null", ""]] = Field(None, q=Exclude())
    time__gt: Optional[time | Literal["null", ""]] = Field(None, q=Exclude())

    def __call__(self):
        q = self._connect_fields()

        if self.date__gt and self.time__gt:
            q &= Q(timestamp__gt=datetime.combine(self.date__gt, self.time__gt))

        return Mymodel.objects.filter(q)
```

**Comment by vitalik** [comment]
would this work ?


```Python
class Filters(FilterSchema):
    date__gt: date = None
    time__gt: time = None

    def custom_expression(self) -> Q:
        q = Q()
        if self.date__gt and self.time__gt:
            q &= Q(timestamp__gt=datetime.combine(self.date__gt, self.time__gt))
        return q
```

**Reply by Fak3** [reply]
No, it won't work in the real environment. The issue is that there are multiple other fields that I want to filter on. So I have to call `self._connect_fields()` which will include date and time fields into queryset. So without my hack using custom list subclass, queryset will produce invalid sql filter.
